src/cow-react/modules/limitOrders/hooks/useFetchInitialPrice.ts

```diff
@@ -0,0 +1,103 @@
+import { useEffect, useState, useCallback } from 'react'
+import { useAtomValue } from 'jotai'
+import { useWeb3React } from '@web3-react/core'
+
+import { getNativePrice } from '@cow/api/gnosisProtocol/api'
+import { WrappedTokenInfo } from 'state/lists/wrappedTokenInfo'
+import { Field } from 'state/swap/actions'
+import { limitRateAtom } from '../state/limitRateAtom'
+import { useLimitOrdersTradeState } from './useLimitOrdersTradeState'
+import { limitDecimals } from '../utils/limitDecimals'
+
+function _getAddress(currency: WrappedTokenInfo): string | null {
+  let address: string | null = null
+
+  if (!currency) {
+    return null
+  }
+
+  if (currency.address) {
+    address = currency.address
+  }
+
+  if (currency.tokenInfo) {
+    address = currency.tokenInfo.address
+  }
+
+  return address
+}
+
+// Fetches the INPUT and OUTPUT price and calculates initial Active rate
+export function useFetchInitialPrice() {
+  const { chainId } = useWeb3React()
+
+  // Global state
+  const { inputCurrency, outputCurrency } = useLimitOrdersTradeState()
+  const { isInversed } = useAtomValue(limitRateAtom)
+
+  // Local state
+  const [inputPrice, setInputPrice] = useState<number | null>(null)
+  const [outputPrice, setOutputPrice] = useState<number | null>(null)
+  const [finalPrice, setFinalPrice] = useState<string | null>(null)
+
+  // Get single price and set the local state
+  const getPrice = useCallback(
+    async (address: string, field: Field) => {
+      // Set INPUT or OUTPUT local state based on field param
+      const setters = { setInputPrice, setOutputPrice }
+      const setPrice = setters[field === Field.INPUT ? 'setInputPrice' : 'setOutputPrice']
+
+      if (!chainId) {
+        return
+      }
+
+      try {
+        const res = await getNativePrice(chainId, address)
+
+        if (res?.price) {
+          setPrice(res?.price)
+        }
+      } catch (err) {
+        setPrice(null)
+        setFinalPrice(null)
+      }
+    },
+    [chainId]
+  )
+
+  // Handle price fetching
+  useEffect(() => {
+    const inputAddress = _getAddress(inputCurrency as WrappedTokenInfo)
```

Comment:
It's not true, currency is not WrappedTokenInfo

Reply:
If I log the value it actually is WrappedTokenInfo but I see your point. Updated 
![Screenshot from 2022-10-27 13-58-18](https://user-images.githubusercontent.com/34926005/198278476-43ec4ffd-926b-49f5-8487-4e1121869af4.png)